### PR TITLE
update stacks2 to version 2.53

### PR DESCRIPTION
**Pull request by bernt-matthias**
- [x] incorporate changes mentioned in the changelog
- [x] added python script to check barcode file (I often get error reports caused by faulty barcode files which were quite hard to debug)

FOR CONTRIBUTOR:
* [x] - I have read the [CONTRIBUTING.md](https://github.com/galaxyproject/tools-iuc/blob/master/CONTRIBUTING.md) document and this tool is appropriate for the tools-iuc repo.
* [x] - License permits unrestricted use (educational + commercial)
* [ ] - This PR adds a new tool or tool collection
* [x] - This PR updates an existing tool or tool collection
* [ ] - This PR does something else (explain below)

### CODE REVIEW

File: tools/stacks2/stacks_populations.xml [hunk 1]
```diff
@@ -548,21 +542,58 @@ $advanced_options.log_fst_comp
...
+        <!-- test w gstacks output as input, wo popmap, and Genetic map output options that have been added in 2.5 -->
+        <test>
+            <param name="options_usage|input_type" value="stacks" />
+            <param name="options_usage|input_aln">
+                <collection type="list">
+                    <element name="catalog.calls.vcf" ftype="vcf" value="gstacks/catalog.calls.vcf" />
+                    <element name="catalog.fa.gz" ftype="fasta.gz" value="gstacks/catalog.fa.gz" />
+               </collection>
+            </param>
+            <section name="genetic_map_options" title="Genetic map output options (population map must specify a genetic cross)" expanded="false">
```
Comment (nsoranzo) [inline tools/stacks2/stacks_populations.xml]:
```suggestion
            <section name="genetic_map_options">
```

File: tools/stacks2/stacks_populations.xml [hunk 2]
```diff
@@ -548,21 +542,58 @@ $advanced_options.log_fst_comp
...
+                <param argument="map_type" value="CP"/>
```
Comment (nsoranzo) [inline tools/stacks2/stacks_populations.xml]:
```suggestion
                <param name="map_type" value="CP"/>
```

File: tools/stacks2/stacks_populations.xml [hunk 3]
```diff
@@ -548,21 +542,58 @@ $advanced_options.log_fst_comp
...
+                <param argument="map_format" value="joinmap" />
```
Comment (nsoranzo) [inline tools/stacks2/stacks_populations.xml]:
```suggestion
                <param name="map_format" value="joinmap" />
```

File: tools/stacks2/stacks_populations.xml [hunk 4]
```diff
@@ -277,27 +268,40 @@ $advanced_options.log_fst_comp
...
-            <param argument="--log_fst_comp" type="boolean" checked="false" truevalue="--log_fst_comp" falsevalue="" label="Log components of Fst/Phi_st calculations to a dataset" />
+            <param argument="--log-fst-comp" type="boolean" checked="false" truevalue="--log-fst-comp" falsevalue="" label="Log components of Fst/Phi_st calculations to a dataset" />
+        </section>
+
+        <!-- Genetic map output options (population map must specify a genetic cross): -->
+        <section name="genetic_map_options" title="Genetic map output options (population map must specify a genetic cross)" expanded="false">
+            <param argument="--map-type" type="select" optional="true" label="genetic map type to write" help="needs to be selected to trigger the additional Genotyping markers output">
+                <option value="CP">CP</option>
+                <option value="DH">DH</option>
+                <option value="F2">BC1</option>
```
Comment (abretaud) [inline tools/stacks2/stacks_populations.xml]: F2 or BC1?

**Comment by nsoranzo** [comment]
It would probably be good to address the (unrelated) warnings from planemo lint:
```
+Linting tool /home/travis/build/galaxyproject/tools-iuc/tools/stacks2/stacks_kmerfilter.xml
Applying linter output... WARNING
.. WARNING: Tool collection output clean_pair doesn't define an output format.
.. WARNING: Tool collection output discarded_pair doesn't define an output format.
...
+Linting tool /home/travis/build/galaxyproject/tools-iuc/tools/stacks2/stacks_clonefilter.xml
Applying linter output... WARNING
.. WARNING: Tool collection output clean_pair doesn't define an output format.
.. WARNING: Tool collection output discarded_pair doesn't define an output format.
```

**Comment by bernt-matthias** [comment]
@nsoranzo good idea. Seems that `format_source` seems not to work. I guess that this is because the inputs conditional https://github.com/galaxyproject/tools-iuc/blob/b0f23c94a1de68b06c31e08e5d2b0c975b36c3dc/tools/stacks2/macros.xml#L192 uses the same name in both when tags... What do you think?

**Comment by nsoranzo** [comment]
@bernt-matthias I've looked at the code and it looks fine to me, not sure what's going wrong.

**Comment by bernt-matthias** [comment]
Maybe its the use of `multiple="true"` for these tools?

**Comment by bernt-matthias** [comment]
Locally with planemo 0.62.1 all tests fail .. If testing against dev I have the same result as here.

**Comment by bernt-matthias** [comment]
Hey @nsoranzo : `format_source` problem is 'solved' by only using the input name without the conditional. See also https://github.com/galaxyproject/galaxy/pull/9493. 

@abretaud  @yvanlebras : I had a little idea for ustacks: currently aĺl samples are processed together in one job. Since each sample can be processed independently one could change this (maybe let the user choose). For separate processing users would need to flatten the produced list of lists in an additional step. Do you thinks its worth the effort?

**Comment by bgruening** [comment]
@bernt-matthias @abretaud what is the status here?

**Comment by bernt-matthias** [comment]
Fine from my side. Would be nice to have some feedback on my idea from @abretaud or @yvanlebras

**Comment by yvanlebras** [comment]
Sorry, miss the last ping. More choice for user is better. We can propose to update training materials with a special point on flattening list of lists. Thank you for your work on stacks and ideas!

**Comment by abretaud** [comment]
For ustacks, I trust you and @yvanlebras more than me, I haven't touched radseq data in years :)
Do you want to merge as is, or wait for this ustacks change and/or #2934?

**Comment by bernt-matthias** [comment]
Stacks 2.53 is ready to go.

File: tools/stacks2/stacks_gstacks.xml [hunk 5]
```diff
@@ -222,11 +222,11 @@ gstacks
                 <has_text text="--write-alignments" />
             </assert_command>
             <output name="output_log" ftype="txt"><assert_contents><has_text text="done." /></assert_contents></output>
-            <output_collection name="gstacks_out" type="list" count="2"/>
-            <output_collection name="gstacks_alns_out" type="list" count="2">
-                <element name="PopA_01" file="gstacks/PopA_01.alns.bam" ftype="bam" />
-                <element name="PopA_02" file="gstacks/PopA_02.alns.bam" ftype="bam" />
-            </output_collection>
+<!--            <output_collection name="gstacks_out" type="list" count="2"/>-->
```
Comment (bgruening) [inline tools/stacks2/stacks_gstacks.xml:225]: should this be really out?

File: tools/stacks2/stacks_gstacks.xml [hunk 6]
```diff
@@ -202,8 +202,8 @@ gstacks
             <output name="output_log" ftype="txt" file="gstacks/gstacks.log" lines_diff="8"/>
             <output name="distribs" ftype="txt" file="gstacks/gstacks.log.distribs" compare="sim_size"/>
             <output_collection name="gstacks_out" type="list" count="2">
-                <element name="catalog.calls.vcf" file="gstacks/catalog.calls.vcf" ftype="vcf" lines_diff="2"/>
-                <element name="catalog.fa.gz" file="gstacks/catalog.fa.gz" ftype="fasta.gz"/>
+                <element name="catalog.calls.vcf" file="gstacks/catalog.calls.vcf" ftype="vcf" lines_diff="4"/>
+                <element name="catalog.fa.gz" file="gstacks/catalog.fa.gz" ftype="fasta.gz" compare="sim_size"/>
```
Comment (bgruening) [inline tools/stacks2/stacks_gstacks.xml:206]: can you use the new size test instead maybe?

File: tools/stacks2/stacks_gstacks.xml (hunk 5 above)
Comment (bernt-matthias) [reply tools/stacks2/stacks_gstacks.xml]: no

File: tools/stacks2/stacks_gstacks.xml (hunk 6 above)
Comment (bernt-matthias) [reply tools/stacks2/stacks_gstacks.xml]: I did where possible and useful

**Comment by bernt-matthias** [comment]
tata. after spliiting the kmerfilter and clonefilter test-data outputs tests are finally running.

**Comment by bernt-matthias** [comment]
Seems like the deployment timed out: https://github.com/galaxyproject/tools-iuc/runs/875092952?check_suite_focus=true

Really wondering if someone gets a note/mail on failing deployment jobs. If not something like this would be really useful.

**Comment by bgruening** [comment]
Retriggered ....

**Comment by bernt-matthias** [comment]
Thanks. Worked this time.

**Pull request merged**